streams/src/main/java/org/apache/kafka/streams/processor/internals/RecordCollectorImpl.java

```diff
@@ -125,41 +204,62 @@ private boolean productionExceptionIsFatal(final Exception exception) {
         return securityException || communicationException;
     }
 
-    private <K, V> void recordSendError(
-        final K key,
-        final V value,
-        final Long timestamp,
-        final String topic,
-        final Exception exception
-    ) {
-        String errorLogMessage = LOG_MESSAGE;
-        String errorMessage = EXCEPTION_MESSAGE;
-
-        if (exception instanceof TimeoutException) {
-            final String topicTimeoutHint = String.format(TIMEOUT_HINT_TEMPLATE, topic);
-            errorLogMessage += topicTimeoutHint;
-            errorMessage += topicTimeoutHint;
-        } else if (exception instanceof RetriableException) {
-            // There is no documented API for detecting retriable errors, so we rely on `RetriableException`
-            // even though it's an implementation detail (i.e. we do the best we can given what's available)
-            errorLogMessage += PARAMETER_HINT;
-            errorMessage += PARAMETER_HINT;
+    private void recordSendError(final String topic, final Exception exception, final ProducerRecord<byte[], byte[]> serializedRecord) {
+        String errorMessage = String.format(SEND_EXCEPTION_MESSAGE, topic, taskId, exception.toString());
+
+        if (productionExceptionIsFatal(exception)) {
```

Comment:
Actually I want to hide the details like whether it is a security error or communication error from the error message (the exception object itself is still included in the stack trace) since I feel such details is better not to be exposed to users.

Reply:
If we are not exposing this detail, then it would be a bit weird to classify them in the first place.

Reply:
You mean `productionExceptionIsFatal` itself? I feel internally we can still such logic (and which may be updated later if necessary) while not exposing it publicly so that we cannot easily change it at will